node/src/lib.rs

```
@@ -226,22 +306,31 @@ impl DriveWrapper {
     }
 
     fn js_create_initial_state_structure(mut cx: FunctionContext) -> JsResult<JsUndefined> {
-        let js_using_transaction = cx.argument::<JsBoolean>(0)?;
+        let js_transaction = cx.argument::<JsValue>(0)?;
+
+        let maybe_boxed_transaction_address = if !js_transaction.is_a::<JsUndefined, _>(&mut cx) {
+            let handle = js_transaction
+                .downcast_or_throw::<JsBox<PlatformWrapperTransactionAddress>, _>(&mut cx)?;
+
+            Some(***handle)
+        } else {
+            None
+        };
+
         let js_callback = cx.argument::<JsFunction>(1)?.root(&mut cx);
 
         let drive = cx
             .this()
-            .downcast_or_throw::<JsBox<DriveWrapper>, _>(&mut cx)?;
-
-        let using_transaction = js_using_transaction.value(&mut cx);
+            .downcast_or_throw::<JsBox<PlatformWrapper>, _>(&mut cx)?;
 
         drive
-            .send_to_drive_thread(move |platform: &Platform, transaction, channel| {
+            .send_to_drive_thread(move |platform: &Platform, transactions, channel| {
+                let transaction_address = maybe_boxed_transaction_address
+                    .expect("transaction address should be available");
+
                 platform
                     .drive
-                    .create_initial_state_structure(
-                        using_transaction.then(|| transaction).flatten(),
-                    )
+                    .create_initial_state_structure(transactions.get(&transaction_address))
```

We need to throw an error in JS thread if transaction_address is incorrect.